addons/account_tax_python/models/account_tax.py

```diff
@@ -47,11 +48,30 @@ def compute_all(self, price_unit, currency=None, quantity=1.0, product=None, par
         for tax in self.filtered(lambda r: r.amount_type == 'code'):
             localdict = self._context.get('tax_computation_context', {})
             localdict.update({'price_unit': price_unit, 'quantity': quantity, 'product': product, 'partner': partner, 'company': company})
-            safe_eval(tax.python_applicable, localdict, mode="exec", nocopy=True)
+            tax.evaluate_expression_code(tax.python_applicable, localdict)
             if localdict.get('result', False):
                 taxes += tax
         return super(AccountTaxPython, taxes).compute_all(price_unit, currency, quantity, product, partner, is_refund=is_refund, handle_price_include=handle_price_include)
 
+    @api.constrains('python_compute', 'python_applicable')
+    def _check_python_compute_and_applicable(self):
+        product = self.env['product.product'].search([], limit=1)
```

Comment:
A bit weird ... And what if someone is configuring a tax on a database without any product yet ? 
We should at least bypass the test in case no product is found. Corner case, indeed, but you'd essentially block people facing it without doing that :/ Not perfect, though.